### PR TITLE
CDAP-15819: Add-field plugin validation.

### PR DESCRIPTION
**Pull request by AntonGensitskiy**
JIRA: https://issues.cask.co/browse/CDAP-15819
WIKI: https://wiki.cask.co/display/CE/Guidelines+on+Plugin+Validation

In scope of this PR:

- integrated with new validation API

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -26,13 +26,10 @@
 
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <cdap.version>6.0.0-SNAPSHOT</cdap.version>
+    <cdap.version>6.1.0-SNAPSHOT</cdap.version>
```
Comment (mlozbin-cybervisiontech) [inline pom.xml:29]: please fix _exportcontents

File: src/main/java/io/cdap/plugin/AddFieldConfig.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+          .withConfigProperty(AS_UUID);
+      }
+    }
+
+    if (inputSchema != null) {
+      for (Schema.Field field : inputSchema.getFields()) {
+        if (field.getName().equals(fieldName)) {
+          failureCollector.addFailure(String.format("Field '%s' already exists in the input schema.", fieldName),
+                                      "Provide field that not present the input schema.")
+            .withConfigProperty(FIELD_NAME);
```
Comment (mlozbin-cybervisiontech) [inline src/main/java/io/cdap/plugin/AddFieldConfig.java]: you should also mark inputSchema field (there is a method to mark it invalid)

File: src/main/java/io/cdap/plugin/MultiFieldAdderConfig.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  }
+
+  public static Builder builder(MultiFieldAdderConfig copy) {
+    return builder()
+      .setFieldValue(copy.fieldValue);
+  }
+
+  Map<String, String> getFieldValue() throws IllegalArgumentException {
+    Map<String, String> values = new TreeMap<>();
+    if (containsMacro("fieldValue") || fieldValue == null || fieldValue.trim().isEmpty()) {
```
Comment (mlozbin-cybervisiontech) [inline src/main/java/io/cdap/plugin/MultiFieldAdderConfig.java]: use constant

File: src/test/java/io/cdap/plugin/MultiFieldAdderConfigTest.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    MockFailureCollector failureCollector = new MockFailureCollector(MOCK_STAGE);
+    MultiFieldAdderConfig config = MultiFieldAdderConfig.builder(VALID_CONFIG)
+      .setFieldValue("test_1:s_value_1,test_2,test_2:s_value_2")
+      .build();
+
+    config.validate(failureCollector);
+    Assert.assertTrue(failureCollector.getValidationFailures().isEmpty());
+  }
+
+  private static void assertValidationFailed(MockFailureCollector failureCollector, List<List<String>> paramNames) {
```
Comment (mlozbin-cybervisiontech) [inline src/test/java/io/cdap/plugin/MultiFieldAdderConfigTest.java]: move all validation helpers to utility class. See https://github.com/data-integrations/whole-file-ingest/pull/7/files#diff-a222ac89d65b50efb72de4d23fb0ae28

**Review by mlozbin-cybervisiontech — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (AntonGensitskiy) [reply pom.xml:29]: Done.

File: src/main/java/io/cdap/plugin/AddFieldConfig.java (hunk 2 above)
Comment (AntonGensitskiy) [reply src/main/java/io/cdap/plugin/AddFieldConfig.java]: Done.

File: src/main/java/io/cdap/plugin/MultiFieldAdderConfig.java (hunk 3 above)
Comment (AntonGensitskiy) [reply src/main/java/io/cdap/plugin/MultiFieldAdderConfig.java]: Done.

File: src/test/java/io/cdap/plugin/MultiFieldAdderConfigTest.java (hunk 4 above)
Comment (AntonGensitskiy) [reply src/test/java/io/cdap/plugin/MultiFieldAdderConfigTest.java]: Done.

**Pull request merged**